docs/user_needs/zephyr_rtos_user.sdoc

```diff
@@ -0,0 +1,87 @@
+[DOCUMENT]
+TITLE: Zephyr RTOS User
+REQ_PREFIX: ZEP-
+
+[GRAMMAR]
+IMPORT_FROM_FILE: user_needs.sgra
+
+[FREETEXT]
+
+TBD
+
+[/FREETEXT]
+
+[USER_NEED]
+UID: ZEP-127
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: OSS License Allows Statically Linked Private Extensions
+STATEMENT: >>>
+As a Product Manufacturer of resource-constrained devices,
+
+I need open-source software to be licensed such that the proprietary code statically linked with the open source software can be kept private.
+<<<
+
+[USER_NEED]
+STATUS: Draft
+TYPE: User Need
+USER: Zephyr RTOS User
+TITLE: Easily Switch Application to different MCU Architecture
+STATEMENT: >>>
+As a Zephyr RTOS user,
+
+I want to be able to easily switch my application to a different MCU architecture (x86, ARM Cortex-M/A, RISCV etc.).
+<<<
+
+[USER_NEED]
+UID: ZEP-128
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Development Systems Options for Windows, Linux
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need embedded software development system options for Windows and Linux host operating systems.
+<<<
+
+[USER_NEED]
+UID: ZEP-129
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: RTOS with Preemptive Scheduling
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need an RTOS that provides preemptive scheduling of threads,
+
+so that I can interleave the priorities of work in my device to meet latency constraints.
+<<<
+
+[USER_NEED]
+UID: ZEP-130
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Sources Configurable for Proprietary HW
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need open source software that configurable for reuse on my proprietary hardware.
```

Comment:
`configurable for reuse` is not understood. Let's make it more clear. Which configurability aspects you would like to bring forward?

Reply:
I'll change "reuse" to "use".

Reply:
... or delete "reuse on" ...

Reply:
1) Proprietary is too limited. Use Bespoke.
2) Aspects: hardware configurability (constraints that have to be passed in and settings). 
3) There is also Feature configurability. This has both enablements and settings. 

 (ISO26262 calls this "configuration", i.e., stuff that happens at configuration/build time). 

Note: Zephyr adopted two specific technologies: kconfig and device tree to enable Configurability.
4) Another aspect that needs to be captured is Reuse (Greg): Without changing a commit of Zephyr repository, we can build any product and get it certified.